backend/src/layer/data/swka_parser/html_parser.rs

```diff
@@ -0,0 +1,468 @@
+//!                     <span class="bgp price_2">4,60 &euro;</span>
+//!                     <span class="bgp price_3">4,20 &euro;</span>
+//!                     <span class="bgp price_4">3,55 &euro;</span>
+//!                     <div style="clear: both;"></div>
+//!                     <a href="javascript:;"
+//!                     title="&Oslash; Umwelt-Score"
+//!                     onclick="toggleRating('5240287810491942285')">
+//!                         <!-- This is the environment score node identified by -->
+//!                         <!--`ENV_SCORE_NODE_CLASS_SELECTOR` it contains an attribute called -->
+//!                         <!--`ENV_SCORE_ATTRIBUTE_NAME`, which contains the environment score -->
+//!                         <div id="average-stars-1551112451474757280"
+//!                             class="enviroment_score average" data-rating="3"
+//!                             data-numstars="3"></div>
+//!                     </a>
+//!                 <tr>
+//! <!-- ... -->
+//! ```
+
+use crate::interface::mensa_parser::{
+    model::{Dish, ParseCanteen, ParseLine},
+    ParseError,
+};
+use crate::util::{Additive, Allergen, Date, MealType, Price};
+use regex::Regex;
+use scraper::element_ref::Text;
+use scraper::{ElementRef, Html, Selector};
+
+const ROOT_NODE_CLASS_SELECTOR: &str = "div.main-content";
+const CANTEEN_NAME_NODE_CLASS_SELECTOR: &str = "h1.mensa_fullname";
+
+const DATE_SUPER_NODE_CLASS_SELECTOR: &str = "ul.canteen-day-nav";
+const DATE_NODE_CLASS_SELECTOR: &str = "a";
+const DATE_ATTRIBUTE_NAME: &str = "rel";
+const DAY_NODE_CLASS_SELECTOR: &str = "div.canteen-day";
+
+const LINE_NODE_CLASS_SELECTOR: &str = "tr.mensatype_rows";
+const LINE_NAME_NODE_CLASS_SELECTOR: &str = "td.mensatype";
+
+const DISH_NODE_CLASS_SELECTOR: &str = "tr.mt-";
+const DISH_TYPE_NODE_CLASS_SELECTOR: &str = "img.mealicon_2";
+const DISH_TYPE_ATTRIBUTE_NAME: &str = "title";
+const DISH_NAME_NODE_CLASS_SELECTOR: &str = "span.bg";
+const DISH_INFO_NODE_CLASS_SELECTOR: &str = "sup";
+const DISH_PRICE_NODE_CLASS_SELECTOR: &str = "span.bgp.price_";
+const ENV_SCORE_NODE_CLASS_SELECTOR: &str = "div.enviroment_score.average";
+const ENV_SCORE_ATTRIBUTE_NAME: &str = "data-rating";
+
+const DATE_FORMAT: &str = "%Y-%m-%d";
+/// A Regex for getting prices in euros. A price consists of 1 or more digits, followed by a comma and then exactly two digits
+const PRICE_REGEX: &str = r"([0-9]*),([0-9]{2})";
+/// A Regex for getting allergens. An allergen consists of a single Uppercase letter followed by one or more upper- or lowercase letters (indicated by \w+)
+const ALLERGEN_REGEX: &str = r"[A-Z]\w+";
+/// A regex for getting additives. An additive consists of one or two digits
+const ADDITIVE_REGEX: &str = r"[0-9]{1,2}";
+
+const NUMBER_OF_MEAL_TYPES: usize = 8;
+
+const SELECTOR_PARSE_E_MSG: &str = "Error while parsing Selector string";
+const REGEX_PARSE_E_MSG: &str = "Error while parsing regex string";
```

Comment:
Sämtliche Regexe und Selectoren sollten einmalig gebaut und Programmstart ausgewertet werden. Vergleicht mal bitte mit zum Beispiel lazy oder once. 
-- Patrick

Reply:
will be changed